grpclog/component.go

```diff
@@ -0,0 +1,236 @@
+/*
+ *
+ * Copyright 2020 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package grpclog
+
+import (
+	"fmt"
+	"math"
+	"os"
+	"strconv"
+	"strings"
+
+	"google.golang.org/grpc/internal/grpclog"
+)
+
+// componentData records the settings for a component.
+type componentData struct {
+	name      string
+	verbosity int
+	level     int
+}
+
+const envName = "GRPC_GO_LOG_LEVEL"
+const sentinel = math.MinInt32
+
+// log levels
+const (
+	levelInfo    = 0
+	levelWarning = 1
+	levelError   = 2
+)
+
+var environmentVars = map[string]*componentData{}
+var prefixVars = map[string]*componentData{}
+var cache = map[string]*componentData{}
+
+// init extracts the component settings from the environment variable.
+func init() {
+	// Pull environment variable data and put in environmentVars and prefixVars
+	v, _ := os.LookupEnv(envName)
+	environmentVars, prefixVars = parseEnvironmentVar(v)
+}
+
+// parseEnvironmentVar parses an environment variable string and pulls the component settings data.
+func parseEnvironmentVar(envVar string) (map[string]*componentData, map[string]*componentData) {
```

Comment:
Are you going to move all changes related to the new env variable to a separate PR?
I think the story we have around the new variable and old one is not right.

We can keep the components in this PR, but only with the prefix behavior. Controlling logging behavior using components can be done later.

Reply:
+1 please remove the environment parsing from this PR and we will follow-up with it in another one.

Reply:
Environment parsing was removed.